detox/src/Detox.js

```diff
@@ -32,75 +32,44 @@ class Detox {
     );
 
     const {artifactsConfig, deviceConfig, session} = config;
+    this._artifactsConfig = artifactsConfig;
     this._deviceConfig = deviceConfig;
     this._userSession = deviceConfig.session || session;
     this._client = null;
     this._server = null;
-    this._artifactsManager = new ArtifactsManager(artifactsConfig);
+    this._artifactsManager = null;
```

Comment:
All is `null` at the beginning, so I know for sure what was initialized and cleaned up at a specific point of time.